tensorflow_serving/example/inception_client.cc

```diff
@@ -0,0 +1,179 @@
+
+    pbuf->pubseekpos(0, std::ios::in);
+    pbuf->sgetn(image, fileSize);
+    imageFile.close();
+    
+
+    proto.set_dtype(::tensorflow::DataType::DT_STRING);
+    proto.add_string_val(image, fileSize);
+
+    std::cout << "image size:" << fileSize << std::endl;
+    proto.mutable_tensor_shape()->add_dim()->set_size(1);
+
+    inputs["images"] = proto;
+    
+    Status status = stub_->Predict(&context, predictRequest, &response);
+    if (status.ok()) {
+      std::cout << "call predict ok" << std::endl;
+      std::cout << "outputs size is "<< response.outputs_size() << std::endl;
+      OutMap& map_outputs =  *response.mutable_outputs();
+      OutMap::iterator iter;
+      int output_index = 0;
+      for(iter = map_outputs.begin();iter != map_outputs.end(); ++iter){
+        std::cout << "output " << output_index << " name is "<< iter->first << std::endl;
+
+        std::cout << "output " << output_index << " type is "<< 
+            ::tensorflow::DataType_Name(
+              iter->second.dtype()) << std::endl;
+
+        int value_size =  0;
+        switch(iter->second.dtype()){
+          case ::tensorflow::DataType::DT_STRING:
+            value_size = iter->second.string_val_size();
+            break;
+          case ::tensorflow::DataType::DT_FLOAT:
+            value_size = iter->second.float_val_size();
+            break;
+          default:
+            break;
+        }
+        std::cout << "outputs has "<< value_size << " value(s)" ;
+
+        std::cout << " and shaped as [ " ;
+        int dim_size = iter->second.tensor_shape().dim_size();
+        ::google::protobuf::int64 * dims = 
+            new ::google::protobuf::int64 [dim_size];
+        for(int d=0; d < dim_size; ++d){
+          dims[d] =  iter->second.tensor_shape().dim(d).size();
+          std::cout << dims[d] << " ";
+        }
+        std::cout << "]" << std::endl;
+
+        for(int k=0; k < value_size; ++k) {
+          switch(iter->second.dtype()){
+            case ::tensorflow::DataType::DT_STRING:
+              std::cout << "value at " << k << " is :" << iter->second.string_val(k) << std::endl;
+              break;
+            case ::tensorflow::DataType::DT_FLOAT:
+              std::cout << "value at " << k << " is :" << iter->second.float_val(k) << std::endl;
+              break;
+            default:
+              break;
+          }
+        }  
+         
+        if(dims) {
+          delete[] dims;
+          dims = NULL;
+        }
+        ++output_index;
+      }
+      if(image) {
+        delete[] image;
+        image = NULL;
+      }
+      return "ok";
+    } else {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      if(image) {
+        delete[] image;
+        image = NULL;
+      }
+      return "RPC failed";
+    }
+  }
+
+ private:
+  std::unique_ptr<PredictionService::Stub> stub_;
+};
+
+
+std::string GetImagePath(int argc, char** argv) {
```

Comment:
Can you please try tensorflow::Flag(...), which will allow you to delete this method?